src/main/java/org/rumbledb/compiler/TranslationVisitor.java

```diff
@@ -605,6 +605,39 @@ public Node visitCastExpr(JsoniqParser.CastExprContext ctx) {
         return new CastExpression(mainExpression, sequenceType, createMetadataFromContext(ctx));
     }
 
+    @Override
+    public Node visitArrowExpr(JsoniqParser.ArrowExprContext ctx) {
+        Expression mainExpression = (Expression) this.visitUnaryExpr(ctx.unaryExpr());
```

Comment:
E=>F(A, B, C)

is simply the same as

F(E, A, B, C)

so no need for new expression machinery.